storage/innobase/log/log0recv.cc

```diff
@@ -4841,6 +4841,53 @@ bool recv_dblwr_t::validate_page(const page_id_t page_id, lsn_t max_lsn,
   goto check_if_corrupted;
 }
 
+dberr_t recv_dblwr_t::recover_encrypted_page(fil_space_t *space,
+                                             uint32_t page_no,
+                                             byte *read_buf) const noexcept
+{
+  for (byte *page : pages)
+  {
+    if (page_get_page_no(page) != page_no
+        || buf_page_is_corrupted(true, page, space->flags))
+      continue;
+    memcpy(read_buf, page, space->physical_size());
+    buf_tmp_buffer_t* slot= buf_pool.io_buf_reserve(false);
+    slot->allocate();
+    if (!fil_space_decrypt(space, slot->crypt_buf, read_buf))
+    {
+      slot->release();
+      continue;
+    }
+
+    if (space->is_compressed())
+    {
+      ulint write_size= fil_page_decompress(slot->crypt_buf, read_buf,
+                                            space->flags);
+      slot->release();
+      if (write_size == 0)
+        continue;
+    }
+    else slot->release();
```

Comment:
This could be simplified:
```c++
    buf_tmp_buffer_t *slot= buf_pool.io_buf_reserve(false);
    ut_a(slot);
    slot->allocate();
    if (!fil_space_decrypt(space, slot->crypt_buf, read_buf) ||
        (space->is_compressed() && 
         !fil_page_decompress(slot->crypt_buf, read_buf, space->flags))
    {
      slot->release();
      continue;
    }
    slot->release();
```
Because `slot->reserve()` is just a store of a constant, it does not seem to be useful to avoid duplicating that, by doing something like
```c++
slot->allocate();
const bool invalid= …;
slot->release();
if (invalid)
  continue;
```